# Business Understanding

The objective of this project is to analyze aviation accident data from the National Transportation Safety Board (NTSB) to provide business recommendations for aircraft purchase. The company is interested in expanding its fleet and wants to minimize the risk associated with aircraft selection.

### Key Questions:
- What are the key factors contributing to aviation accidents?
- Which aircraft models are associated with fewer accidents?
- How can data-driven insights help the company make informed purchasing decisions?

### Stakeholders:
- **Aviation Company**: Looking for data-driven insights on low-risk aircraft.
- **Data Science Team**: Responsible for conducting the analysis.


# Data Understanding

The dataset used for this analysis contains records of aviation accidents from 1962 to 2023, provided by the NTSB. It includes information on accident details, aircraft types, locations, and other key attributes.

### Data Source:
- The dataset is sourced from the **National Transportation Safety Board (NTSB)**.



In [74]:
import pandas as pd

# Load the dataset
df = pd.read_csv('C:/Users/User/OneDrive/Desktop/moringa/project phase 1/Aircraft-risk-analysis/data/raw data/AviationData.csv')

# Show the first few rows of the dataset
df.head()


C:\Users\User\AppData\Local\Temp\ipykernel_32412\149211319.py:4: DtypeWarning: Columns (6,7,28) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv('C:/Users/User/OneDrive/Desktop/moringa/project phase 1/Aircraft-risk-analysis/data/raw data/AviationData.csv')


Event.Id Investigation.Type Accident.Number  Event.Date  \
0  20001218X45444           Accident      SEA87LA080  1948-10-24   
1  20001218X45447           Accident      LAX94LA336  1962-07-19   
2  20061025X01555           Accident      NYC07LA005  1974-08-30   
3  20001218X45448           Accident      LAX96LA321  1977-06-19   
4  20041105X01764           Accident      CHI79FA064  1979-08-02   

          Location        Country   Latitude  Longitude Airport.Code  \
0  MOOSE CREEK, ID  United States        NaN        NaN          NaN   
1   BRIDGEPORT, CA  United States        NaN        NaN          NaN   
2    Saltville, VA  United States  36.922223 -81.878056          NaN   
3       EUREKA, CA  United States        NaN        NaN          NaN   
4       Canton, OH  United States        NaN        NaN          NaN   

  Airport.Name  ... Purpose.of.flight Air.carrier Total.Fatal.Injuries  \
0          NaN  ...          Personal         NaN                  2.0   
1          NaN  ...          Personal         NaN                  4.0   
2          NaN  ...          Personal         NaN                  3.0   
3          NaN  ...          Personal         NaN                  2.0   
4          NaN  ...          Personal         NaN                  1.0   

  Total.Serious.Injuries Total.Minor.Injuries Total.Uninjured  \
0                    0.0                  0.0             0.0   
1                    0.0                  0.0             0.0   
2                    NaN                  NaN             NaN   
3                    0.0                  0.0             0.0   
4                    2.0                  NaN             0.0   

  Weather.Condition  Broad.phase.of.flight   Report.Status Publication.Date  
0               UNK                 Cruise  Probable Cause              NaN  
1               UNK                Unknown  Probable Cause       19-09-1996  
2               IMC                 Cruise  Probable Cause       26-02-2007  
3               IMC                 Cruise  Probable Cause       12-09-2000  
4               VMC               Approach  Probable Cause       16-04-1980  

[5 rows x 31 columns]

In [75]:
# Check basic information about the dataset
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 88889 entries, 0 to 88888
Data columns (total 31 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Event.Id                88889 non-null  object 
 1   Investigation.Type      88889 non-null  object 
 2   Accident.Number         88889 non-null  object 
 3   Event.Date              88889 non-null  object 
 4   Location                88837 non-null  object 
 5   Country                 88663 non-null  object 
 6   Latitude                34382 non-null  object 
 7   Longitude               34373 non-null  object 
 8   Airport.Code            50132 non-null  object 
 9   Airport.Name            52704 non-null  object 
 10  Injury.Severity         87889 non-null  object 
 11  Aircraft.damage         85695 non-null  object 
 12  Aircraft.Category       32287 non-null  object 
 13  Registration.Number     87507 non-null  object 
 14  Make                    88826 non-null

In [76]:
# Describe the dataset to understand the statistical properties
df.describe()

Number.of.Engines  Total.Fatal.Injuries  Total.Serious.Injuries  \
count       82805.000000          77488.000000            76379.000000   
mean            1.146585              0.647855                0.279881   
std             0.446510              5.485960                1.544084   
min             0.000000              0.000000                0.000000   
25%             1.000000              0.000000                0.000000   
50%             1.000000              0.000000                0.000000   
75%             1.000000              0.000000                0.000000   
max             8.000000            349.000000              161.000000   

       Total.Minor.Injuries  Total.Uninjured  
count          76956.000000     82977.000000  
mean               0.357061         5.325440  
std                2.235625        27.913634  
min                0.000000         0.000000  
25%                0.000000         0.000000  
50%                0.000000         1.000000  
75%                0.000000         2.000000  
max              380.000000       699.000000

## Data Cleaning and Formatting

To simplify the analysis, we selected only the columns relevant to understanding aviation accident patterns. These include identifiers, location data, aircraft details, flight purpose, injury severity, and weather conditions.

### Selected Columns:
- `Event.Id`, `Accident.Number`, `Event.Date`: Basic identifiers and timestamps
- `Location`, `Country`, `Latitude`, `Longitude`: Geographical information
- `Injury.Severity`, `Total.Fatal.Injuries`, `Total.Serious.Injuries`, `Total.Minor.Injuries`, `Total.Uninjured`: Impact analysis
- `Aircraft.damage`, `Aircraft.Category`, `Make`, `Model`: Aircraft details
- `Number.of.Engines`, `Engine.Type`: Engine characteristics
- `Purpose.of.flight`: Operational context
- `Weather.Condition`, `Broad.phase.of.flight`: Environmental and situational factors


In [77]:
# Getting the colums in the dataset to determine the columns needed
df.columns.tolist()


['Event.Id',
 'Investigation.Type',
 'Accident.Number',
 'Event.Date',
 'Location',
 'Country',
 'Latitude',
 'Longitude',
 'Airport.Code',
 'Airport.Name',
 'Injury.Severity',
 'Aircraft.damage',
 'Aircraft.Category',
 'Registration.Number',
 'Make',
 'Model',
 'Amateur.Built',
 'Number.of.Engines',
 'Engine.Type',
 'FAR.Description',
 'Schedule',
 'Purpose.of.flight',
 'Air.carrier',
 'Total.Fatal.Injuries',
 'Total.Serious.Injuries',
 'Total.Minor.Injuries',
 'Total.Uninjured',
 'Weather.Condition',
 'Broad.phase.of.flight',
 'Report.Status',
 'Publication.Date']

In [78]:
columns_to_keep = [
    'Event.Id',
    'Accident.Number',
    'Event.Date',
    'Location',
    'Country',
    'Latitude',
    'Longitude',
    'Injury.Severity',
    'Aircraft.damage',
    'Aircraft.Category',
    'Make',
    'Model',
    'Number.of.Engines',
    'Engine.Type',
    'Purpose.of.flight',
    'Total.Fatal.Injuries',
    'Total.Serious.Injuries',
    'Total.Minor.Injuries',
    'Total.Uninjured',
    'Weather.Condition',
    'Broad.phase.of.flight'
]

df_clean= df[columns_to_keep]
df_clean.info()
df_clean.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 88889 entries, 0 to 88888
Data columns (total 21 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Event.Id                88889 non-null  object 
 1   Accident.Number         88889 non-null  object 
 2   Event.Date              88889 non-null  object 
 3   Location                88837 non-null  object 
 4   Country                 88663 non-null  object 
 5   Latitude                34382 non-null  object 
 6   Longitude               34373 non-null  object 
 7   Injury.Severity         87889 non-null  object 
 8   Aircraft.damage         85695 non-null  object 
 9   Aircraft.Category       32287 non-null  object 
 10  Make                    88826 non-null  object 
 11  Model                   88797 non-null  object 
 12  Number.of.Engines       82805 non-null  float64
 13  Engine.Type             81793 non-null  object 
 14  Purpose.of.flight       82697 non-null

Event.Id Accident.Number  Event.Date         Location        Country  \
0  20001218X45444      SEA87LA080  1948-10-24  MOOSE CREEK, ID  United States   
1  20001218X45447      LAX94LA336  1962-07-19   BRIDGEPORT, CA  United States   
2  20061025X01555      NYC07LA005  1974-08-30    Saltville, VA  United States   
3  20001218X45448      LAX96LA321  1977-06-19       EUREKA, CA  United States   
4  20041105X01764      CHI79FA064  1979-08-02       Canton, OH  United States   

    Latitude  Longitude Injury.Severity Aircraft.damage Aircraft.Category  \
0        NaN        NaN        Fatal(2)       Destroyed               NaN   
1        NaN        NaN        Fatal(4)       Destroyed               NaN   
2  36.922223 -81.878056        Fatal(3)       Destroyed               NaN   
3        NaN        NaN        Fatal(2)       Destroyed               NaN   
4        NaN        NaN        Fatal(1)       Destroyed               NaN   

   ...     Model Number.of.Engines    Engine.Type Purpose.of.flight  \
0  ...     108-3               1.0  Reciprocating          Personal   
1  ...  PA24-180               1.0  Reciprocating          Personal   
2  ...      172M               1.0  Reciprocating          Personal   
3  ...       112               1.0  Reciprocating          Personal   
4  ...       501               NaN            NaN          Personal   

  Total.Fatal.Injuries  Total.Serious.Injuries  Total.Minor.Injuries  \
0                  2.0                     0.0                   0.0   
1                  4.0                     0.0                   0.0   
2                  3.0                     NaN                   NaN   
3                  2.0                     0.0                   0.0   
4                  1.0                     2.0                   NaN   

   Total.Uninjured  Weather.Condition Broad.phase.of.flight  
0              0.0                UNK                Cruise  
1              0.0                UNK               Unknown  
2              NaN                IMC                Cruise  
3              0.0                IMC                Cruise  
4              0.0                VMC              Approach  

[5 rows x 21 columns]

In [79]:
#check for mising data
df_clean.isnull().sum()

Event.Id                      0
Accident.Number               0
Event.Date                    0
Location                     52
Country                     226
Latitude                  54507
Longitude                 54516
Injury.Severity            1000
Aircraft.damage            3194
Aircraft.Category         56602
Make                         63
Model                        92
Number.of.Engines          6084
Engine.Type                7096
Purpose.of.flight          6192
Total.Fatal.Injuries      11401
Total.Serious.Injuries    12510
Total.Minor.Injuries      11933
Total.Uninjured            5912
Weather.Condition          4492
Broad.phase.of.flight     27165
dtype: int64

## How i wil handle the missing data

After analyzing the data and comparing the number of rows and missing data this is what i have decided to go with

### Columns to Drop:
- **Latitude** and **longitude**: Over 61% of data missing, making it unreliable for analysis. 
- **Aircraft.Category**: More than 63% of data missing. Dropped due to the lack of sufficient data.
- **Broad.phase.of.flight**:due to a significant portion of missing data.

### Columns to Keep and Replace Missing Data:
- **Location**: 52 missing values, replaced with "Unknown"
- **Country**: 226 missing values, replaced with "Unknown"
- **Injury.Severity**: 1,000 missing values, replaced with the mode.
- **Make and Model**: Small number of missing values , replaced with "Unknown"
- **Number.of.Engines and Engine.Type**: Significant missing values, replaced with "Unknown" .
- **Purpose.of.flight**: 6,192 missing values, replaced with "Unknown".
- **Total Injuries** (Fatal, Serious, Minor) and **Total Uninjured**: (zero for no injuries).
- **Weather.Condition**: Replaced with "Unknown".
- **Aircraft damage**: Replaced with "Unknown".

In [80]:
# Dropping columns with excessive missing data (Latitude, Longitude, Aircraft.Category, Broad.phase.of.flight)
df_clean.drop(columns=['Latitude', 'Longitude', 'Aircraft.Category', 'Broad.phase.of.flight'], inplace=True)


# Replacing missing data based on the decisions:

# 1. Location and Country - Replace missing values with 'Unknown'
df_clean['Location'].fillna('Unknown', inplace=True)
df_clean['Country'].fillna('Unknown', inplace=True)

# 2. Injury Severity - Replace missing values with the mode (most frequent value)
df_clean['Injury.Severity'].fillna(df_clean['Injury.Severity'].mode()[0], inplace=True)

# 3. Make and Model - Replace missing values with 'Unknown'
df_clean['Make'].fillna('Unknown', inplace=True)
df_clean['Model'].fillna('Unknown', inplace=True)

# 4. Number of Engines and Engine Type - Replace missing values with 'Unknown'
df_clean['Number.of.Engines'].fillna('Unknown', inplace=True)
df_clean['Engine.Type'].fillna('Unknown', inplace=True)

# 5. Purpose of flight - Replace missing values with 'Unknown'
df_clean['Purpose.of.flight'].fillna('Unknown', inplace=True)

# 6. Total injuries and Total Uninjured - Impute missing values with zero (assuming no injuries reported if not available)
df_clean['Total.Fatal.Injuries'].fillna(0, inplace=True)
df_clean['Total.Serious.Injuries'].fillna(0, inplace=True)
df_clean['Total.Minor.Injuries'].fillna(0, inplace=True)
df_clean['Total.Uninjured'].fillna(0, inplace=True)
# 7. Weaer Condition - Replace missing values with 'Unknown'
df_clean['Weather.Condition'].fillna('Unknown', inplace=True)

#8. Aircraft damage - replacing with unknown
df_clean['Aircraft.damage'].fillna('Unknown', inplace=True)



C:\Users\User\AppData\Local\Temp\ipykernel_32412\1038811968.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_clean.drop(columns=['Latitude', 'Longitude', 'Aircraft.Category', 'Broad.phase.of.flight'], inplace=True)
C:\Users\User\AppData\Local\Temp\ipykernel_32412\1038811968.py:8: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_clean['Location'].fillna('Unknown', inplace=True)
C:\Users\User\AppData\Local\Temp\ipykernel_32412\1038811968.py:9: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable

In [81]:
print(f"Cleaned Data Shape: {df_clean.shape}")
print("Missing Values After Cleaning:")
print(df_clean.isnull().sum())

Cleaned Data Shape: (88889, 17)
Missing Values After Cleaning:
Event.Id                  0
Accident.Number           0
Event.Date                0
Location                  0
Country                   0
Injury.Severity           0
Aircraft.damage           0
Make                      0
Model                     0
Number.of.Engines         0
Engine.Type               0
Purpose.of.flight         0
Total.Fatal.Injuries      0
Total.Serious.Injuries    0
Total.Minor.Injuries      0
Total.Uninjured           0
Weather.Condition         0
dtype: int64
